src/schema/v2/OrderedSet/deleteOrderedSetItemMutation.ts

```diff
@@ -0,0 +1,86 @@
+import { GraphQLString, GraphQLUnionType, GraphQLObjectType } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { GraphQLNonNull } from "graphql"
+import { ResolverContext } from "types/graphql"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import { OrderedSetItemType } from "../item"
+
+interface Input {
+  id: string
+  itemId: string
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "deleteOrderedSetItemSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    setItem: {
+      type: OrderedSetItemType,
+      resolve: (orderedSetItem) => orderedSetItem,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "deleteOrderedSetItemFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "deleteOrderedSetItemResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const deleteOrderedSetItemMutation = mutationWithClientMutationId<
+  Input,
+  any | null,
+  ResolverContext
+>({
+  name: "deleteOrderedSetItemMutation",
+  description: "deletes an item to an ordered set.",
+  inputFields: {
+    id: { type: new GraphQLNonNull(GraphQLString) },
+    itemId: { type: new GraphQLNonNull(GraphQLString) },
+  },
+  outputFields: {
+    orderedSetItemOrError: {
+      type: ResponseOrErrorType,
+      description: "On success: the ordered set item deleted.",
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async (
+    { id, itemId },
+    { deleteSetItemLoader, setLoader }
+  ) => {
+    if (!deleteSetItemLoader) {
+      throw new Error(
+        "You need to pass a X-Access-Token header to perform this action"
+      )
+    }
+
+    try {
+      const res = await deleteSetItemLoader({ id, itemId })
+
+      const { item_type } = await setLoader(id)
```

Comment:
Tricksy! Clearly `itemType` shouldn't really _need_ to be a required argument to this mutation (the set and item id should be sufficient), as sets can only be associated with one item type. The gravity API doesn't need `item_type` either in this endpoint.

Maybe we should make `itemType` a required argument to this mutation (being passed through to here, not sent to Gravity)? I've often thought it would be nice for artist JSON, artwork JSON, etc. in Gravity to all have a consistent way of identifying the type from the response (vs. the brittle inference of `birthday` being present, for instance).

Reply:
My first approach was to use a required argument, but then I changed that in df1adf6. What changed my mind was that it seemed like a nice implementation detail to remove this as a client concern; I agree that this seemed like something that I would have expected to be coming back from gravity with the JSON response. Also, if we do end up having gravity return out the type, this would limit code changes to `mp`, and we wouldn't need to also update the client(s). 

All that being said, I definitely hear you on this not feeling conventional and will happily revert back, especially if you think I'm overlooking potential client concerns.